plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerTypeMapping.java

```diff
@@ -69,6 +72,107 @@ public void testVarbinary()
                 .execute(getQueryRunner(), trinoCreateAsSelect("test_varbinary"));
     }
 
+    @Test
+    public void testTime()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("time(0)", "'00:00:00'", createTimeType(0), "TIME '00:00:00'")
+                .addRoundTrip("time(6)", "'00:00:00.000000'", createTimeType(6), "TIME '00:00:00.000000'")
+                .addRoundTrip("time(6)", "'00:00:00.123456'", createTimeType(6), "TIME '00:00:00.123456'")
+                .addRoundTrip("time(0)", "'12:34:56'", createTimeType(0), "TIME '12:34:56'")
+                .addRoundTrip("time(6)", "'12:34:56.123456'", createTimeType(6), "TIME '12:34:56.123456'")
+
+                // maximal value for a precision
+                .addRoundTrip("time(0)", "'23:59:59'", createTimeType(0), "TIME '23:59:59'")
+                .addRoundTrip("time(1)", "'23:59:59.9'", createTimeType(1), "TIME '23:59:59.9'")
+                .addRoundTrip("time(2)", "'23:59:59.99'", createTimeType(2), "TIME '23:59:59.99'")
+                .addRoundTrip("time(3)", "'23:59:59.999'", createTimeType(3), "TIME '23:59:59.999'")
+                .addRoundTrip("time(4)", "'23:59:59.9999'", createTimeType(4), "TIME '23:59:59.9999'")
+                .addRoundTrip("time(5)", "'23:59:59.99999'", createTimeType(5), "TIME '23:59:59.99999'")
+                .addRoundTrip("time(6)", "'23:59:59.999999'", createTimeType(6), "TIME '23:59:59.999999'")
+                .addRoundTrip("time(7)", "'23:59:59.9999999'", createTimeType(7), "TIME '23:59:59.9999999'")
+
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_time"))
+                .execute(getQueryRunner(), trinoCreateAsSelect(getSession(), "test_time"));
+
+        SqlDataTypeTest.create()
```

Comment:
Why can't it be merged into the previous one (all tests are same just the runner is different).

This deserves a comment if kept this way.

Reply:
I added a comment.

`trinoCreateAndInsert` requires type in literals, otherwise, SQL Server recognizes them as varchars in the INSERT statement. On the other hand, `sqlServerCreateAndInsert` passes this literal `TIME 'xx:xx:xx'` to SQL Server which doesn't recognize this syntax. We might pull it off with `CAST('xx:xx:xx' AS TIME)`. WDYT?

Reply:
I do not feel strongly about merging those case.

Another thing you can do is to merge two added test sections with 
```
                .execute(getQueryRunner(), trinoCreateAndInsert(getSession(), "test_time"))
                .execute(getQueryRunner(), trinoCreateAsSelect(getSession(), "test_time"));
```
and drop 
```
                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_time"))
```
from the preexistent section

Reply:
I'm fine with not merging. It was just not obvious to me why they aren't merged. Now I know. Thanks. 🙂

Reply:
@losipiuk I totally missed that I added the second set of tests myself. It makes so much sense to group them, thanks!